packages/jest-matchers/src/index.js

```diff
@@ -87,21 +103,18 @@ const makeThrowingMatcher = (
 
     _validateResult(result);
 
-    if ((result.pass && isNot) || (!result.pass && !isNot)) { // XOR
-      let message = result.message;
-
-      // for performance reasons some of the messages are evaluated
-      // lazily
-      if (typeof message === 'function') {
-        message = message();
-      }
-
-      if (!message) {
-        message = utils.RECEIVED_COLOR(
-          'No message was specified for this matcher.',
-        );
-      }
+    if ((result.pass && !isNot) || (!result.pass && isNot)) {
+      global[GLOBAL_STATE].state.passedExpectations.push({
+        actual: '',
+        expected: '',
+        message: getMessage(result.message),
+        passed: true,
+      });
+    }
+    
 
+    if ((result.pass && isNot) || (!result.pass && !isNot)) { // XOR
```

Comment:
can we consolidate two XORs into if/else?
or assign it to a variable `passed` or something? :)